python/private/coverage_deps.bzl

```diff
@@ -0,0 +1,165 @@
+"""Dependencies for coverage.py used by the hermetic toolchain.
+"""
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+load(
+    "//python:versions.bzl",
+    "MINOR_MAPPING",
+    "PLATFORMS",
+)
+
+# Update with './tools/update_coverage_deps.py <version>'
+#START: managed by update_coverage_deps.py script
+_coverage_deps = {
```

Comment:
Ah hm. Something I just realized is we run the risk of conflicting with a user installed coverage library. e.g.
the toolchain uses coverage X
The user, in their requirements, uses coverage Y

At runtime, we'll have 2 different coverage libraries available, and it's not well defined which will be used.

I'm not sure what we can do about this, though. I guess add an arg to toolchain creation and dependency setup that allows the user to specify the coverage target to use, and skip having one automatically installed? Or...IDK. The key attribute we're looking for is that the toolchain and user binary reference the same target (or equiv; that we only end up with 1 coverage library in runfiles).

Reply:
That would be an argument in favor of using `py_library` as the target type in the remote for `coverage_dep`, so things other than the toolchain could use it.

I don't think there's a way around having the coverage tool used by the toolchain being potentially different from what's declared in requirements, because you can't actually do a `pip install` until the toolchain is configured.  I don't think it's a dealbreaker, though; it might be a little surprising to find out that coverage is running with a version of `coveragepy` distinct from what was in your `requirements.txt`, but at the same time I don't think it'll actually _interefere_ with anything.

Reply:
For now, can you default `register_coverage_deps=False`? Otherwise we'll potentially cause issues with people who have coverage in their deps already, and we don't really have any solution or workaround available to them.

Yeah, we were kicking around ideas on how to best address this. I'm not sure there is much to be done for this in this PR. The only thing I can think of would be to change the  `register_coverage_tool` arg from a bool to something that allows the caller to pass in their own target.

> you can't actually do a pip install until the toolchain is configured. 

Hmmm, well, what if you _could_? If we had a separate "pip tools" or "requirements resolution" toolchain, then maybe we'd be able to do that. Just thinking out loud.

The two more appealing ideas we came up with were:
 * Making sure the toolchains' coverage was last in the import order (probably by modifying the bootstrap, when the `bootstrap_template` arg becomes available). That would at least give well-defined order for which version gets used and lets the user have more control over the version used.
 * Have the toolchain only have the coverarge tool entry point, not the coverage library itself; each test is responsible for bringing the necessary deps for the coverage tool (and try to figure out how to make that as easy/transparent to test writers as possible)